pkg/config/msp.go

```diff
@@ -74,3 +82,29 @@ func addRootCAToOrgMSP(rootCA *x509.Certificate, org *cb.ConfigGroup) error {
 
 	return nil
 }
+
+// RevokeCertificateFromMSP takes an x509 certificate and adds it to the
+// revocation list for the specified application org MSP.
+func RevokeCertificateFromMSP(cert *x509.Certificate, orgName string, config *cb.Config) error {
+	org := getOrgsFromApplication(config)[orgName]
+	mspConfig, fabricMSPConfig, err := getMSPConfig(org)
+	if err != nil {
+		return fmt.Errorf("getting msp config: %v", err)
+	}
+
+	// TODO validate that this certificate was issued by this MSP
+
+	buffer := bytes.NewBuffer(nil)
+	err = pem.Encode(buffer, &pem.Block{Type: "CERTIFICATE", Bytes: cert.Raw})
+	if err != nil {
+		return err
+	}
+	fabricMSPConfig.RevocationList = append(fabricMSPConfig.RevocationList, buffer.Bytes())
```

Comment:
the bytes of a CRL is not an x509 certificate, but a CRL that looks like:

> type CertificateList struct {
> 	TBSCertList        TBSCertificateList
> 	SignatureAlgorithm AlgorithmIdentifier
> 	SignatureValue     asn1.BitString
> }

Reply:
@yacovm What is the slice of byte slices for the fabricMSPConfig.RevocationList represented by in the CRL?

Reply:
Left a TODO to address this. Stomach is yelling at me to eat some lunch now. :)

Reply:
@stephyee , it's just either an array of either PEM encoded or DER ASN1 encoded CRLs. Look at the [code that reads them](https://github.com/hyperledger/fabric/blob/master/msp/mspimplsetup.go#L197-L215), and you then can make the inverse code that writes them.